### PR TITLE
conflict only when item present in both A and B

### PR DESCRIPTION
**Pull request by ravishetye**
related to https://vertaai.slack.com/archives/CJK8B4TN1/p1588070408027000

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/versioning/blob/diff/ConflictGenerator.java [hunk 1]
```diff
@@ -151,7 +151,7 @@ private static AutogenVersionEnvironmentDiff getVersionDiff(
     Map<String, AutogenPythonRequirementEnvironmentBlob> mapC = getRequirementMap(c);
     HashSet<String> keys = new HashSet<>();
     keys.addAll(mapA.keySet());
-    keys.addAll(mapB.keySet());
+    keys.retainAll(mapB.keySet());
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/versioning/blob/diff/ConflictGenerator.java:154]: Is this doing intersections instead of union?

Comment (ravishetye) [reply backend/src/main/java/ai/verta/modeldb/versioning/blob/diff/ConflictGenerator.java:154]: yes

**Review by conradoverta — APPROVED** [review]
LGTM assuming you have validated and checked these are the only locations where this happens.

File: backend/src/test/java/ai/verta/modeldb/MergeTest.java [hunk 2]
```diff
@@ -395,6 +395,28 @@ static Blob getDatasetBlobFromPath(String path, long size) {
         .build();
   }
 
+  static Blob getDatasetBlobFromPathMultiple(String path, long size) {
```
Comment (anandjakhaniya) [inline backend/src/test/java/ai/verta/modeldb/MergeTest.java:398]: I don't know the purpose of this. otherwise LGTM.

**Review by anandjakhaniya — APPROVED** [review]

**Comment by anandjakhaniya** [comment]
Not relevant to this PR but I see two same conditions at https://github.com/VertaAI/modeldb/pull/593/files#diff-2f1798b3ef5ea47202599842c1780034R64-R71 so just adding a comment

**Comment by anandjakhaniya** [comment]
I think the system was crashing with a null pointer from https://github.com/VertaAI/modeldb/pull/593/files#diff-2f1798b3ef5ea47202599842c1780034R428 maybe

File: backend/src/test/java/ai/verta/modeldb/MergeTest.java [hunk 3]
```diff
@@ -395,6 +395,28 @@ static Blob getDatasetBlobFromPath(String path, long size) {
...
+                    PathDatasetBlob.newBuilder()
+                        .addComponents(
+                            PathDatasetComponentBlob.newBuilder()
+                                .setPath(path)
+                                .setSize(size)
+                                .setLastModifiedAtSource(time)
+                                .build())
+                        .addComponents(
+                            PathDatasetComponentBlob.newBuilder()
+                                .setPath(path + "~")
```
Comment (ravishetye) [inline backend/src/test/java/ai/verta/modeldb/MergeTest.java:412]: it just adds a new component .

File: backend/src/test/java/ai/verta/modeldb/MergeTest.java (hunk 2 above)
Comment (ravishetye) [reply backend/src/test/java/ai/verta/modeldb/MergeTest.java:398]: util  function to generate the bug report case

**Pull request merged**